File: frontend/lib/models/moodboard.dart
import 'image.dart';

class Moodboard {
  final int id;
  final String title;
  final String description;
  final String? coverImageUrl;
  final List<Image> images;

  Moodboard({
    required this.id,
    required this.title,
    required this.description,
    this.coverImageUrl,
    this.images = const [],
  });

  factory Moodboard.fromJson(Map<String, dynamic> json) {
    var imageList = json['images'] as List? ?? [];
    List<Image> images = imageList.map((i) => Image.fromJson(i)).toList();

    return Moodboard(
      id: json['id'],
      title: json['title'],
      description: json['description'],
      coverImageUrl: json['coverImageUrl'],
      images: images,
    );
  }
} 

File: backend/src/main/java/com/filmtool/api/model/Moodboard.java
package com.filmtool.api.model;

import java.util.ArrayList;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonManagedReference;

import jakarta.persistence.*;

@Entity
@Table(name = "moodboards")
public class Moodboard {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    @Column
    private String description;

    @Column
    private String coverImageUrl;

    @OneToMany(mappedBy = "moodboard", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonManagedReference
    private List<Image> images = new ArrayList<>();

    // Default constructor for JPA
    public Moodboard() {
    }

    // Getters and Setters
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getCoverImageUrl() {
        return coverImageUrl;
    }

    public void setCoverImageUrl(String coverImageUrl) {
        this.coverImageUrl = coverImageUrl;
    }

    public List<Image> getImages() {
        return images;
    }

    public void setImages(List<Image> images) {
        this.images = images;
    }

    @Override
    public String toString() {
        return "Moodboard{" +
                "id=" + id +
                ", title='" + title + '\'' +
                ", description='" + description + '\'' +
                ", coverImageUrl='" + coverImageUrl + '\'' +
                ", images=" + (images != null ? images.size() : 0) +
                '}';
    }
} 